src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -533,6 +550,29 @@ public static IEnumerable<object[]> Copy_TestData()
         }
 
         [Theory]
+        public static void ForEach()
+        {
+            Array.ForEach<int>(new int[] { }, new Action<int>(i => { throw new InvalidOperationException(); }));
+            // Did not throw; no items
+
+            int counter = 0;
+            Array.ForEach<int>(new int[] { 1, 2, 3 }, new Action<int>(i => counter += i));
+            Assert.Equal(6, counter);
```

Comment:
This validates that the delegate was executed for all of the items, but not the order in which it was executed.  It'd be nice if the test checked that, too.